e2e/playwright/tests/@version-history-pagination/test.spec.ts

```diff
@@ -0,0 +1,165 @@
+  const startTime = Date.now();
+  for (let i = 0; i < testNumOfVersions; i++) {
+    const commandStartTime = Date.now();
+    execSync(`kubectl kots set config ${testAppSlug} -n ${testNamespace} item_1="version-${i}"`);
+    const commandEndTime = Date.now();
+    const commandDuration = (commandEndTime - commandStartTime) / 1000; // Convert to seconds
+    const totalDuration = (commandEndTime - startTime) / 1000; // Convert to seconds
+    console.log(`creating version ${i+1} of ${testNumOfVersions} (took ${commandDuration.toFixed(2)}s, average ${(totalDuration / (i + 1)).toFixed(2)}s)`);
+  }
+  const endTime = Date.now();
+  const totalDuration = (endTime - startTime) / 1000; // Convert to seconds
+  console.log(`total time to create ${testNumOfVersions} versions: ${totalDuration.toFixed(2)}s`);
+
+  // validate that the versions created via the CLI are visible via the CLI with pagination
+  const versionCheckConfig: VersionCheckConfig = {
+    testAppSlug: testAppSlug,
+    testNamespace: testNamespace,
+    testDefaultPageSize: testDefaultPageSize,
+    testLatestSequence: testLatestSequence,
+  };
+
+  checkVersions(versionCheckConfig);
+
+  // validate that the versions created via the CLI are visible via the UI
+  console.log("validating the first page of versions via the UI");
+  await page.getByRole('link', { name: 'Version history' }).click();
+  await expect(page.getByText('All versions')).toBeVisible();
+  // should be 251 (new version available at top of page)
+  await expect(page.getByTestId('available-updates-card').getByTestId('version-sequence')).toBeVisible();
+  await expect(page.getByTestId('available-updates-card').getByTestId('version-sequence')).toContainText(`Sequence ${testLatestSequence}`);
+  // should be 251 (first entry in all versions card)
+  await expect(page.getByTestId('all-versions-card').getByTestId('version-history-row-0').getByTestId('version-sequence')).toBeVisible();
+  await expect(page.getByTestId('all-versions-card').getByTestId('version-history-row-0').getByTestId('version-sequence')).toContainText(`Sequence ${testLatestSequence}`);
+  // should be 250 (second entry in all versions card)
+  await expect(page.getByTestId('version-history-row-1').getByTestId('version-sequence')).toBeVisible();
+  await expect(page.getByTestId('version-history-row-1').getByTestId('version-sequence')).toContainText(`Sequence ${testLatestSequence - 1}`);
+
+  // should be 232 (last entry in first page of all versions card)
+  await expect(page.getByTestId('version-history-row-19').getByTestId('version-sequence')).toBeVisible();
+  await expect(page.getByTestId('version-history-row-19').getByTestId('version-sequence')).toContainText(`Sequence ${testLatestSequence - 19}`);
+  await expect(page.getByTestId('all-versions-card')).toContainText(`Showing releases 1 - 20 of ${testNumOfVersions + 1}`);
+
+  console.log("validating the second page of versions via the UI");
+  await page.getByText('Next').click();
```

Comment:
should probably have a data-testid, this word can easily exist in the page.

Reply:
I unfortunately wasn't able to find it in the code anywhere

Reply:
i think it's this https://github.com/replicatedhq/kots/blob/d60aabb0347d6a76b5d4fac4b85e7c343bc35b87/web/src/components/shared/Pager.jsx#L85